include/openssl/crypto.h.in

```diff
@@ -93,21 +93,26 @@ int CRYPTO_atomic_load(uint64_t *val, uint64_t *ret, CRYPTO_RWLOCK *lock);
 /* No longer needed, so this is a no-op */
 #define OPENSSL_malloc_init() while(0) continue
 
-void *ERR_raise_malloc_failure(void *ptr,
+void *ERR_raise_malloc_failure(void *(*alloc_fn)(size_t, const char *, int),
+                               size_t num,
                                const char *file, int line, const char *func);
+void *ERR_raise_realloc_failure(void *addr, size_t num,
+                                const char *file, int line, const char *func);
+void *ERR_raise_clear_realloc_failure(void *addr, size_t old_num, size_t num,
+                                      const char *file, int line,
+                                      const char *func);
```

Comment:
Almost good, except we should call the functions differently. Probably `CRYPTO_malloc_with_err()`, `CRYPTO_realloc_with_err()`, and `CRYPTO_clear_realloc_with_err()` would do.

Reply:
Good point - done.